cmd/kogito/command/deploy/deploy_service.go

```diff
@@ -265,3 +317,69 @@ func (i *deployCommand) Exec(cmd *cobra.Command, args []string) (err error) {
 
 	return nil
 }
+
+// matchKogitoResources will verify if the given parameter matches one of the allowed extensions
+// Accepted values are, supported file extensions as it is, directory or url
+// See supportedExtensions env
+// returns the file content as io.Reader if the provided file or url is valid.
+func matchKogitoResources(resource string) (io.Reader, *buildv1.BinaryBuildRequestOptions, error) {
+	log := context.GetDefaultLogger()
+	options := &buildv1.BinaryBuildRequestOptions{}
+	var fileName string
+	// from url, do only basic check, for example, if the url has the suffix that's match the allowed ones
+	switch {
+	case strings.HasPrefix(resource, "http"):
+		parsedURL, err := url.ParseRequestURI(resource)
+		if err != nil || parsedURL.Scheme == "" || parsedURL.Host == "" {
+			return nil, nil, &url.Error{
+				URL: resource,
+				Err: err,
+			}
+		}
+
+		ff := strings.Split(parsedURL.Path, "/")
+		fileName = strings.Join(strings.Fields(ff[len(ff)-1]), "")
+
+		if buildutil.IsSuffixSupported(fileName) {
+			response, err := http.Get(resource)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to download %s, error message: %s", resource, err.Error())
+			}
+			options.AsFile = fileName
+			log.Infof(message.KogitoAppFoundAsset, fileName)
+			return response.Body, options, nil
+		}
+
+	// handle single file, files on dir.
+	default:
+		fileInfo, err := os.Stat(resource)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if fileInfo.Mode().IsRegular() {
+			if buildutil.IsSuffixSupported(resource) {
+				log.Infof(message.KogitoAppFoundFile, resource)
+				ff := strings.Split(resource, "/")
+				fileName = strings.Join(strings.Fields(ff[len(ff)-1]), "")
+				fileReader, err := os.Open(resource)
+				if err != nil {
+					return nil, nil, err
+				}
+				options.AsFile = fileName
+				return fileReader, options, nil
+			}
+
+		} else if fileInfo.Mode().IsDir() {
+			log.Info(message.KogitoAppProvidedFileIsDir)
+
+			ioTgzR, err := buildutil.ProduceTGZfile(resource)
+			if err != nil {
+				return nil, nil, err
+			}
+			return ioTgzR, options, nil
+		}
+		return nil, nil, &kogitoerror.KogitoAssetNotFound{Asset: resource}
+	}
+	return nil, nil, nil
```

Comment:
```suggestion
		if buildutil.IsSuffixSupported(fileName) {
			response, err := http.Get(resource)
			if err != nil {
				return nil, nil, fmt.Errorf("failed to download %s, error message: %s", resource, err.Error())
			}
			options.AsFile = fileName
			log.Infof(message.KogitoAppFoundAsset, fileName)
			return response.Body, options, nil
		}
		return nil, nil, nil

	// handle single file, files on dir.
	default:
		fileInfo, err := os.Stat(resource)
		if err != nil {
			return nil, nil, err
		}

		if fileInfo.Mode().IsRegular() {
			if buildutil.IsSuffixSupported(resource) {
				log.Infof(message.KogitoAppFoundFile, resource)
				ff := strings.Split(resource, "/")
				fileName = strings.Join(strings.Fields(ff[len(ff)-1]), "")
				fileReader, err := os.Open(resource)
				if err != nil {
					return nil, nil, err
				}
				options.AsFile = fileName
				return fileReader, options, nil
			}

		} else if fileInfo.Mode().IsDir() {
			log.Info(message.KogitoAppProvidedFileIsDir)

			ioTgzR, err := buildutil.ProduceTGZfile(resource)
			if err != nil {
				return nil, nil, err
			}
			return ioTgzR, options, nil
		}
		return nil, nil, &kogitoerror.KogitoAssetNotFound{Asset: resource}
	}
```

Reply:
The proposed change is to move `return nil, nil, nil` into line 352 to make switch case flow more readable (all other branches have own return statements so the only way how to reach the final return statement is from that 352 line).

Reply:
Make sense, will apply and test it tomorrow morning.